src/cdp/cdpClient.ts

```diff
@@ -123,18 +114,33 @@ class CdpClientImpl extends EventEmitter {
     return new Promise((resolve, reject) => {
       const id = this._nextId++;
       this._commandCallbacks.set(id, { resolve, reject });
-      const messageObj = { id, method, params };
-      const messageStr = JSON.stringify(messageObj);
+      let messageObj: CdpMessage = { id, method, params };
+      if (this._sessionId) messageObj.sessionId = this._sessionId;
 
-      logCdp('sent > ' + messageStr);
-      this._transport.sendMessage(messageStr);
+      const messageStr = JSON.stringify(messageObj);
+      this._router.sendMessage(messageStr);
     });
   }
 
-  private _onCdpMessage(messageStr: string): void {
-    logCdp('received < ' + messageStr);
+  /**
+   *
+   * Returns a new CdpClient object attached to the given sessionId. The new CdpClient shares the same
+   * underlying transport as this CdpClient.
+   * @param sessionId The sessionId to attach the new client to. Use Target.attachToTarget to obtain a sessionId.
+   */
+  attachToSession(sessionId: string): CdpClient {
```

Comment:
> Another thing: Maybe it would make sense to keep a single cdpClient per sessionId and serve it? It would allow to avoid a commandId collisions.

I updated our code to do this. It looks like Puppeteer does something similar too. I'll update the PR description with the details.